analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/InvalidProjectFunctionalTest.java

```diff
@@ -52,8 +54,16 @@ public void ensureInvalidException() throws IOException, URISyntaxException {
             TestUtils.align(projectRoot, projectRoot.getName(), true);
             fail("No exception thrown");
         } catch (ManipulationUncheckedException e) {
-            assertThat(e.getMessage()).contains(
-                    "For configuration compile; unable to resolve all dependencies: [io.undertow:undertow-core:2.0+, org.apache.commons:commons-lang3:3.8.1.redhat-3]");
+            String regex = ".*For configuration compileClasspath, unable to resolve all project dependencies: \\[(.*)].*";
+            Pattern pattern = Pattern.compile(regex, Pattern.DOTALL);
+            Matcher matcher = pattern.matcher(e.getMessage());
+            assertThat(matcher.matches()).isTrue();
+            assertThat(matcher.groupCount()).isEqualTo(1);
+            String[] dependencies = matcher.group(1).split(", ");
+            String[] expectedDependencies = { "io.undertow:undertow-core:2.0+",
+                    "org.apache.commons:commons-lang3:3.8.1.redhat-3",
+                    "org.springframework:spring-context:5.1.6.INVALID_VERSION" };
+            assertThat(dependencies).containsExactly(expectedDependencies);
```

Comment:
Can you explain why the test check is now different please?

Reply:
1. There's the addition of "org.springframework:spring-context:5.1.6.INVALID_VERSION" which was always there, but didn't show up until we switched all to `implementation` (for Gradle 7). This shows now as configuration `compileClasspath` instead of `compile`.

2. There was a problem with the ordering when switching between Gradle 7 and other versions. With `HashSet`, we don't seem to get a guaranteed order (even if it usually works, since it's not "random"). This code was changed originally to be able to handle changes in order by using `contains` instead of a simple string comparison. However, since I changed certain spots to maintain the original set order, this change should no longer be necessary (but, I still found it to be more flexible).

Reply:
Ok